python/pyarrow/pandas_compat.py

```diff
@@ -623,7 +623,21 @@ def _can_definitely_zero_copy(arr):
     metadata.update(pandas_metadata)
     schema = schema.with_metadata(metadata)
 
-    return arrays, schema
+    # If dataframe is empty but with RangeIndex ->
+    # remember the length of the indexes
+    n_rows = None
+    if len(arrays) == 0 and schema is not None:
```

Comment:
Is the `schema is not None` needed? Above, there is a block `if schema is None` which then creates a schema, which seems to indicate that at this point a `schema` object will always be `not None`?

Reply:
Oh yes, you are right. Removed it.